lms/templates/student_account/register.underscore

```diff
@@ -29,11 +29,17 @@
             <%
             _.each( context.providers, function( provider) {
                 if ( provider.registerUrl ) { %>
-                    <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
-                        <span class="icon fa <%- provider.iconClass %>" aria-hidden="true"></span>
-                        <span aria-hidden="true"><%- provider.name %></span>
-                        <span class="sr"><%- _.sprintf( gettext("Create account using %(providerName)s."), {providerName: provider.name} ) %></span>
-                    </button>
+                    <div class="button-container" <% if ( provider.buttonColor ) { %>style="color: <%- provider.buttonColor %>; background-color: <%- provider.buttonColor %>"<% } %>>
+                        <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
+                            <div class="icon <% if ( provider.iconClass ) { %>fa <%- provider.iconClass %><% } %>" <% if ( provider.buttonColor ) { %>style="background: <%- provider.buttonColor %>"<% } %> aria-hidden="true">
+                                <% if ( provider.iconImage ) { %>
+                                    <img src="<%- provider.iconImage %>" alt="<%- provider.name %> icon" />
+                                <% } %>
+                            </div>
+                            <span <% if ( provider.textColor ) { %>style="color: <%- provider.textColor %>"<% } %> aria-hidden="true"><%- provider.name %></span>
```

Comment:
Yes, this sounds like a good approach to me as well.  Let's leave the door open for @saleem-latif to respond re: custom sass via theming, because it might end up being something you can do quickly/easily -- however no need to block on that so I would plan to move ahead with what you are suggesting.